Source/Core/Core/Boot/Boot_BS2Emu.cpp

```diff
@@ -242,6 +242,30 @@ bool CBoot::EmulatedBS2_GC(const DiscIO::VolumeDisc& volume)
   return RunApploader(/*is_wii*/ false, volume);
 }
 
+static DiscIO::Region CodeRegion(char c)
+{
+  switch (c)
+  {
+  case 'J':  // Japan
+  case 'T':  // Taiwan
+    return DiscIO::Region::NTSC_J;
+  case 'B':  // Brazil
+  case 'M':  // Middle East
+  case 'R':  // Argentina
+  case 'S':  // ???
+  case 'U':  // USA
+  case 'W':  // ???
+    return DiscIO::Region::NTSC_U;
+  case 'A':  // Australia
+  case 'E':  // Europe
+    return DiscIO::Region::PAL;
+  case 'K':  // Korea
+    return DiscIO::Region::NTSC_K;
+  default:
+    return DiscIO::Region::Unknown;
+  }
+}
+
 bool CBoot::SetupWiiMemory(IOS::HLE::IOSC::ConsoleType console_type)
 {
   static const std::map<DiscIO::Region, const RegionSetting> region_settings = {
```

Comment:
Can these serial mappings be changed? "LJ" isn't a valid prefix for JPN, and "LE" isn't a valid one for PAL. Japanese and PAL serial number prefixes have three chars. I'd suggest LJH and LEH (so it's identical to KOR for which it is LKH).

Reply:
Or, maybe, for out-of-region games, just take the serial prefix from the settings.txt and just edit the 2nd character to "match" the region?

Reply:
> Can these serial mappings be changed? As far as I know, "LJ" isn't a valid prefix for JPN, and "LE" isn't a valid one for PAL. Japanese and PAL serial number prefixes have three chars. I'd suggest LJH and LEH (so it's identical to KOR for which it is LKH).

I have now created PR #8678 for this.

> Or, maybe, for out-of-region games, just take the serial prefix from the settings.txt and just edit the 2nd character to "match" the region?

We need values to fall back to in case there isn't already a setting.txt file. (Most users don't import a NAND.)

Also, I don't see how this would solve the problem of having the wrong number of characters.

Reply:
That would solve the problem for the users who did import a NAND, but the solution in #8678 is probably better anyways because it applies to everyone.